### ISSUE TITLE
PyTorch not recognising CUDA on GTX980

### ISSUE THREAD

**Issue by Dreamest** [pytorch/pytorch#50534]
<h1>Description</h1>
PyTorch doesn't seem to recognise CUDA properly despite it working a few days before.

<h1>Code</h1>

```
>> nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2019 NVIDIA Corporation
Built on Sun_Jul_28_19:07:16_PDT_2019
Cuda compilation tools, release 10.1, V10.1.243
```

```
>> import torch 
>>print(torch.__version__)
1.7.1
```

```
>> !nvidia-smi
Thu Jan 14 13:49:24 2021       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 450.102.04   Driver Version: 450.102.04   CUDA Version: 11.0     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  GeForce GTX 980     Off  | 00000000:01:00.0  On |                  N/A |
|  0%   43C    P8    16W / 196W |    214MiB /  4040MiB |      5%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A       973      G   /usr/lib/xorg/Xorg                 75MiB |
|    0   N/A  N/A      1319      G   /usr/bin/gnome-shell               79MiB |
|    0   N/A  N/A      2586      G   ...AAAAAAAA== --shared-files       54MiB |
+-----------------------------------------------------------------------------+
```

```
>> torch.version.cuda
'10.2'
```
```
>> torch.cuda.is_available()
/home/*******/anaconda3/lib/python3.8/site-packages/torch/cuda/__init__.py:52: UserWarning: CUDA initialization: CUDA unknown error - this may be due to an incorrectly set up environment, e.g. changing env variable CUDA_VISIBLE_DEVICES after program start. Setting the available devices to be zero. (Triggered internally at  /opt/conda/conda-bld/pytorch_1607369981906/work/c10/cuda/CUDAFunctions.cpp:100.)
  return torch._C._cuda_getDeviceCount() > 0
False
```

I'm at a loss. CUDA used to work just fine and I don't recall touching the settings, it works properly on Windows, but won't recognize CUDA on Ubuntu. 

cc @ngimel

**Comment by ptrblck** [comment]
The error points to an issue with the CUDA initialization, which might be raised e.g. if your system has updated drivers automatically. Could you restart the machine and check, if this might help?

**Comment by Dreamest** [comment]
Thanks for the quick reply!

I've tried restarting numerous times, and even installing different(both older and newer) nvidia drivers, but all my attempts were to no avail.

**Comment by ngimel** [comment]
Can you compile cuda samples from the cuda toolkit? Do they work?

**Comment by Dreamest** [comment]
Followign the [CUDA Samples](https://github.com/NVIDIA/cuda-samples#cuda-112) git(I hope that's the one you referred to), it fails the 'make' command.

```
make[1]: Entering directory '/home/********/samples/cuda-samples/Samples/memMapIPCDrv'
>>> WARNING - libcuda.so not found, CUDA Driver is not installed.  Please re-install the driver. <<<
[@] /usr/local/cuda/bin/nvcc -ccbin g++ -I../../Common -m64 --threads 0 -gencode arch=compute_35,code=sm_35 -gencode arch=compute_37,code=sm_37 -gencode arch=compute_50,code=sm_50 -gencode arch=compute_52,code=sm_52 -gencode arch=compute_60,code=sm_60 -gencode arch=compute_61,code=sm_61 -gencode arch=compute_70,code=sm_70 -gencode arch=compute_75,code=sm_75 -gencode arch=compute_80,code=sm_80 -gencode arch=compute_86,code=sm_86 -gencode arch=compute_86,code=compute_86 -o helper_multiprocess.o -c ../../Common/helper_multiprocess.cpp
[@] /usr/local/cuda/bin/nvcc -ccbin g++ -I../../Common -m64 --threads 0 -gencode arch=compute_35,code=sm_35 -gencode arch=compute_37,code=sm_37 -gencode arch=compute_50,code=sm_50 -gencode arch=compute_52,code=sm_52 -gencode arch=compute_60,code=sm_60 -gencode arch=compute_61,code=sm_61 -gencode arch=compute_70,code=sm_70 -gencode arch=compute_75,code=sm_75 -gencode arch=compute_80,code=sm_80 -gencode arch=compute_86,code=sm_86 -gencode arch=compute_86,code=compute_86 -o memMapIpc.o -c memMapIpc.cpp
[@] /usr/local/cuda/bin/nvcc -ccbin g++ -m64 -gencode arch=compute_35,code=sm_35 -gencode arch=compute_37,code=sm_37 -gencode arch=compute_50,code=sm_50 -gencode arch=compute_52,code=sm_52 -gencode arch=compute_60,code=sm_60 -gencode arch=compute_61,code=sm_61 -gencode arch=compute_70,code=sm_70 -gencode arch=compute_75,code=sm_75 -gencode arch=compute_80,code=sm_80 -gencode arch=compute_86,code=sm_86 -gencode arch=compute_86,code=compute_86 -o memMapIPCDrv helper_multiprocess.o memMapIpc.o
[@] mkdir -p ../../bin/x86_64/linux/release
[@] cp memMapIPCDrv ../../bin/x86_64/linux/release
[@] /usr/local/cuda/bin/nvcc -ccbin g++ -I../../Common -m64 --threads 0 -gencode arch=compute_35,code=sm_35 -gencode arch=compute_37,code=sm_37 -gencode arch=compute_50,code=sm_50 -gencode arch=compute_52,code=sm_52 -gencode arch=compute_60,code=sm_60 -gencode arch=compute_61,code=sm_61 -gencode arch=compute_70,code=sm_70 -gencode arch=compute_75,code=sm_75 -gencode arch=compute_80,code=sm_80 -gencode arch=compute_86,code=sm_86 -gencode arch=compute_86,code=compute_86 -o memMapIpc_kernel64.fatbin -fatbin memMapIpc_kernel.cu
[@] mkdir -p data
[@] cp -f memMapIpc_kernel64.fatbin ./data
[@] mkdir -p ../../bin/x86_64/linux/release
[@] cp -f memMapIpc_kernel64.fatbin ../../bin/x86_64/linux/release
make[1]: Leaving directory '/home/chenz/samples/cuda-samples/Samples/memMapIPCDrv'
make[1]: Entering directory '/home/chenz/samples/cuda-samples/Samples/simpleCUBLAS'
/usr/local/cuda/bin/nvcc -ccbin g++ -I../../Common  -m64    --threads 0 -gencode arch=compute_35,code=compute_35 -o simpleCUBLAS.o -c simpleCUBLAS.cpp
make[1]: /usr/local/cuda/bin/nvcc: Command not found
make[1]: *** [Makefile:308: simpleCUBLAS.o] Error 127
make[1]: Leaving directory '/home/chenz/samples/cuda-samples/Samples/simpleCUBLAS'
make: *** [Makefile:45: Samples/simpleCUBLAS/Makefile.ph_build] Error 2
```

Stating that libcuda.so is not found.

**Comment by angrypiggy** [comment]
Same problem encountered using 3090. I did not remember touching anything. Did you resolve this issue?

**Comment by Dreamest** [comment]
> Same problem encountered using 3090. I did not remember touching anything. Did you resolve this issue?

No. Was advised to remove everything CUDA related and try, but haven't gotten to it yet.

**Comment by angrypiggy** [comment]
So basically uninstall the cuda and reinstall? My `nvidia-smi` gives me the cuda version 11.2, and `nvcc -V` gives me 10.2. I searched online and found that they do not necessarily need to match. Also, my command to install torch was using cuda 11.0 which is the highest version supported. I was suspecting this could be the issue as well.

**Comment by Dreamest** [comment]
> So basically uninstall the cuda and reinstall? My `nvidia-smi` gives me the cuda version 11.2, and `nvcc -V` gives me 10.2. I searched online and found that they do not necessarily need to match. Also, my command to install torch was using cuda 11.0 which is the highest version supported. I was suspecting this could be the issue as well.

In theory. Haven't tried yet, can't confirm it'll work or screw things up for worse.

**Comment by angrypiggy** [comment]
Thanks, I will keep you updated on the problem status.

**Comment by angrypiggy** [comment]
`sudo apt install nvidia-utils-460` resolved my issue.

**Comment by Dreamest** [comment]
> `sudo apt install nvidia-utils-460` resolved my issue.

Bless your heart, this seem to have solved it for me as well(Although needed to also do 

> sudo apt install libnvidia-compute-460
